### ISSUE TITLE
Fail Catalyst-Plugin-Cache-0.10

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10604]
Migrated from [rt.perl.org#77678](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=77678) (status was 'rejected')

Searchable as RT77678$

**Comment by p5pRT** [comment]
### From brewsterbear@gmail.com

#### Created by brewsterbear@gmail.com



make test failed&#8203;:

\[info\] \*\*\* Request 2 \(2\.000/s\) \[4150\] \[Sun Sep  5 10&#8203;:41&#8203;:50 2010\] \*\*\*
\[debug\] "GET" request for "dvd/by\_id/2" from "127\.0\.0\.1"
\[debug\] Arguments are "dvd/by\_id/2"
\[debug\] Response Code&#8203;: 404; Content\-Type&#8203;: text/html; charset=utf\-8; Content\-Length&#8203;: 13
\[info\] Request took 0\.002573s \(388\.651/s\)
\.\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\+\-\-\-\-\-\-\-\-\-\-\-\.
| Action                                                     | Time      |
\+\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\+\-\-\-\-\-\-\-\-\-\-\-\+
| /default                                                   | 0\.000113s |
| /end                                                       | 0\.000175s |
'\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\+\-\-\-\-\-\-\-\-\-\-\-'


\#   Failed test 'GET dvd object'
\#   at t/31\.test\_rest\.t line 23\.
\# 404
\# Not Found
\# Looks like you failed 1 test of 2\.
t/31\.test\_rest\.t \.\.\.\.\.\. Dubious\, test returned 1 \(wstat 256\, 0x100\)
Failed 1/2 subtests
t/pod\-coverage\.t \.\.\.\.\.\. skipped&#8203;: Author test
t/pod\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\. ok

Test Summary Report
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
t/31\.test\_rest\.t    \(Wstat&#8203;: 256 Tests&#8203;: 2 Failed&#8203;: 1\)
&nbsp;  Failed test&#8203;:  2
&nbsp;  Non\-zero exit status&#8203;: 1
Files=9\, Tests=60\, 10 wallclock secs \( 0\.02 usr  0\.00 sys \+ 10\.01 cusr  0\.44 csys = 10\.47 CPU\)
Result&#8203;: FAIL
Failed 1/9 test programs\. 1/60 subtests failed\.
make&#8203;: \*\*\* \[test\_dynamic\] Error 255

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low

Site configuration information for perl 5.10.0:

Configured by Debian Project at Thu Oct  1 22:36:47 UTC 2009.

Summary of my perl5 (revision 5 version 10 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.6.24-23-server, archname=x86_64-linux-gnu-thread-multi
    uname='linux crested 2.6.24-23-server #1 smp wed apr 1 22:14:30 utc 2009 x86_64 gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=x86_64-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.10 -Darchlib=/usr/lib/perl/5.10 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.10.0 -Dsitearch=/usr/local/lib/perl/5.10.0 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -DDEBUGGING=-g -Doptimize=-O2 -Duseshrplib -Dlibperl=libperl.so.5.10.0 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion='', gccversion='4.4.1', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /lib64 /usr/lib64
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.10.1.so, so=so, useshrplib=true, libperl=libperl.so.5.10.0
    gnulibc_version='2.10.1'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.10.0:
    /etc/perl
    /usr/local/lib/perl/5.10.0
    /usr/local/share/perl/5.10.0
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.10
    /usr/share/perl/5.10
    /usr/local/lib/site_perl
    .


Environment for perl 5.10.0:
    HOME=/root
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From brewsterbear@gmail.com

Incorrect test output in original report\, make test failure below&#8203;:

PERL\_DL\_NONLAZY=1 /usr/bin/perl "\-MExtUtils&#8203;::Command&#8203;::MM" "\-e" 
"test\_harness\(0\, 'inc'\, 'blib/lib'\, 'blib/arch'\)" t/\*\.t
t/basic\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\. 1/? Couldn't construct 
Catalyst&#8203;::Plugin&#8203;::Cache&#8203;::Backend&#8203;::Memory with either list style or hash 
ref style param passing&#8203;:
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
\# Looks like your test exited with 255 just after 1\.
t/basic\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\. Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
All 1 subtests passed
t/config\_backend\_class\.t \.\. 1/? Couldn't construct MyCache with either 
list style or hash ref style param passing&#8203;:
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
\# Tests were run but no plan was declared and done\_testing\(\) was not seen\.
t/config\_backend\_class\.t \.\. Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
All 1 subtests passed
t/config\_guess\_backend\.t \.\. ok
t/currying\_conf\.t \.\.\.\.\.\.\.\.\. 1/? Couldn't construct SomeClass with either 
list style or hash ref style param passing&#8203;:
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
\# Tests were run but no plan was declared and done\_testing\(\) was not seen\.
t/currying\_conf\.t \.\.\.\.\.\.\.\.\. Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
All 1 subtests passed
t/key\_regexes\.t \.\.\.\.\.\.\.\.\.\.\. 1/? Couldn't construct 
Catalyst&#8203;::Plugin&#8203;::Cache&#8203;::Backend&#8203;::Memory with either list style or hash 
ref style param passing&#8203;:
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
&nbsp;         Undefined subroutine &Class&#8203;::MOP&#8203;::is\_class\_loaded called at 
/usr/local/share/perl/5\.10\.0/Catalyst/Utils\.pm line 279\.
\# Tests were run but no plan was declared and done\_testing\(\) was not seen\.
t/key\_regexes\.t \.\.\.\.\.\.\.\.\.\.\. Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
All 2 subtests passed
t/live\_app\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\. \[warn\] Catalyst&#8203;::Plugin&#8203;::Cache config not 
found\, using empty config\!
t/live\_app\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\. 1/5 \[warn\] Catalyst&#8203;::Plugin&#8203;::Cache config 
not found\, using empty config\!
t/live\_app\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\. ok

Test Summary Report


---
t/basic\.t               \(Wstat&#8203;: 65280 Tests&#8203;: 1 Failed&#8203;: 0\)
&nbsp;   Non\-zero exit status&#8203;: 255
t/config\_backend\_class\.t \(Wstat&#8203;: 65280 Tests&#8203;: 1 Failed&#8203;: 0\)
&nbsp;   Non\-zero exit status&#8203;: 255
&nbsp;   Parse errors&#8203;: No plan found in TAP output
t/currying\_conf\.t       \(Wstat&#8203;: 65280 Tests&#8203;: 1 Failed&#8203;: 0\)
&nbsp;   Non\-zero exit status&#8203;: 255
&nbsp;   Parse errors&#8203;: No plan found in TAP output
t/key\_regexes\.t         \(Wstat&#8203;: 65280 Tests&#8203;: 2 Failed&#8203;: 0\)
&nbsp;   Non\-zero exit status&#8203;: 255
&nbsp;   Parse errors&#8203;: No plan found in TAP output
Files=6\, Tests=14\,  1 wallclock secs \( 0\.04 usr  0\.00 sys \+  1\.00 cusr  
0\.05 csys =  1\.09 CPU\)
Result&#8203;: FAIL
Failed 4/6 test programs\. 0/14 subtests failed\.
make&#8203;: \*\*\* \[test\_dynamic\] Error 255

**Comment by p5pRT** [comment]
### From @chorny

Hello\.

If you entered all information correctly into perlbug\, you should have
seen warning "Module Catalyst\-Plugin\-Cache is not a core module\."\.
This warning is correct\. perlbug should be used only for bugs in core
Perl modules\.

Please use Catalyst\-Plugin\-Cache bug tracker for such bug reports&#8203;:
https://rt.cpan.org/Public/Dist/Display.html?Name=Catalyst-Plugin-Cache
\. Link to it is available from this page&#8203;:
http&#8203;://search\.cpan\.org/dist/Catalyst\-Plugin\-Cache/

2010/9/5 brewsterbear@&#8203;gmail\.com \<perlbug\-followup@&#8203;perl\.org>&#8203;:
> Flags&#8203;:
>    category=library
>    severity=low
> \-\-\-
> Site configuration information for perl 5\.10\.0&#8203;:


\-\- 
Alexandr Ciornii\, http&#8203;://chorny\.net

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @nwc10

On Mon\, Sep 06\, 2010 at 03&#8203;:04&#8203;:50PM \+0300\, Alexandr Ciornii wrote&#8203;:
> Hello\.
> 
> If you entered all information correctly into perlbug\, you should have
> seen warning "Module Catalyst\-Plugin\-Cache is not a core module\."\.
> This warning is correct\. perlbug should be used only for bugs in core
> Perl modules\.

I believe that feature to report "not a core module" was only added to perlbug
in 5\.10\.1\.

The reporter was using 5\.10\.0

However\, the documentation for perlbug in 5\.10\.0 does explain that it's for
the core \(and modules shipped with the core\) only\.

But all the above is details\. The key part is&#8203;:

> Please use Catalyst\-Plugin\-Cache bug tracker for such bug reports&#8203;:
> https://rt.cpan.org/Public/Dist/Display.html?Name=Catalyst-Plugin-Cache
> \. Link to it is available from this page&#8203;:
> http&#8203;://search\.cpan\.org/dist/Catalyst\-Plugin\-Cache/

Nicholas Clark

**Comment by p5pRT** [comment]
@chorny - Status changed from 'open' to 'rejected'
